### ISSUE TITLE
Connection description migration breaks on MySQL 8

### ISSUE THREAD

**Issue by ashb** [apache/airflow#12588]
The migration added in #10873 doesn't work on Mysql8 -- it is too long for a text column with utf8 collation.

```
Running upgrade 2c6edca13270 -> 61ec73d9401f, Add description field to connection
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 255, in execute
    self.errorhandler(self, exc, value)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
    raise errorvalue
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 252, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 378, in _query
    db.query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 280, in query
    _mysql.connection.query(self, query)
_mysql_exceptions.OperationalError: (1118, 'Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs')
```

**Apache Airflow version**:

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know**:

**Comment by potiuk** [comment]
Looks like https://github.com/apache/airflow/commit/bf6da166a974331e8cfe416c29cde0206c86d251 -> description should be Text rather than String . How should we approach this? Shall we change the existing migration ? I guess there is not much choice? And I wonder why it did not fail on 5.7 ?  We have utf8mb4 there as well.

**Comment by potiuk** [comment]
More than that - seem that String is treated differently there. There are more big Srings there already.

**Comment by ashb** [comment]
That migration was in a release (even if just beta) so we can't really change it.

New migration needed

**Comment by ashb** [comment]
@potiuk Happy to tackle this if you aren't.

**Comment by potiuk** [comment]
New migration would not help - it would fail in the middle by incrementally running migrations.

**Comment by ashb** [comment]
Oh yeah, I guess change old migration to TEXT, but also create an ALTER migration to make it text (if we were going this route)

**Comment by jackemuk-es** [comment]
@ashb If you change the old migration to TEXT, then the migration fails on postgres. 

PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

**Comment by ashb** [comment]
> @ashb If you change the old migration to TEXT, then the migration fails on postgres. 
> 
> PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

Fails with what error?

**Comment by potiuk** [comment]
Working on a fix in #12596
